query/transformation.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package query contains all the logic to parse
+// and execute queries against the underlying metric system.
+package query
+
+import (
+	"math"
+
+	"github.com/square/metrics/api"
+)
+
+// A transformation takes the list of values, and a "scale" factor:
+// It is the unitless quotient (sample duration / 1s)
+// So if the resolution is "once every 5 minutes" them we get a scale factor of 5*60 = 300
+type transformation func([]float64, transformationParameter) []float64
+
+type transformationParameter struct {
+	scale     float64
+	parameter float64
```

Comment:
hmm.

I'm thinking about passing:

* some subset of `EvaluationContext` (everything in it is too much, because we don't want transforms to be doing a lot of crazy things, such as fetching backend. I think `TimeRange` is good to be passed around.

* if the other PR - https://github.com/square/metrics/pull/43 - is merged, maybe we can pass in the list of values?

i like the fact that the first parameter is `[]float64` because it restricts what we can do with it.